tests/util_tests/test_pgvector_migrator.py

```diff
@@ -0,0 +1,806 @@
+
+
+async def afetch(engine: AlloyDBEngine, query: str) -> Sequence[RowMapping]:
+    async def run(engine, query):
+        async with engine._pool.connect() as conn:
+            result = await conn.execute(text(query))
+            result_map = result.mappings()
+            result_fetch = result_map.fetchall()
+        return result_fetch
+
+    return await engine._run_as_async(run(engine, query))
+
+
+def get_env_var(key: str, desc: str) -> str:
+    v = os.environ.get(key)
+    if v is None:
+        raise ValueError(f"Must set env var {key} to: {desc}")
+    return v
+
+
+@pytest.mark.asyncio
+class TestPgvectorengine:
+    @pytest.fixture(scope="module")
+    def db_project(self) -> str:
+        return get_env_var("PROJECT_ID", "project id for google cloud")
+
+    @pytest.fixture(scope="module")
+    def db_region(self) -> str:
+        return get_env_var("REGION", "region for AlloyDB instance")
+
+    @pytest.fixture(scope="module")
+    def db_cluster(self) -> str:
+        return get_env_var("CLUSTER_ID", "cluster for AlloyDB instance")
+
+    @pytest.fixture(scope="module")
+    def db_instance(self) -> str:
+        return get_env_var("INSTANCE_ID", "instance for alloydb")
+
+    @pytest.fixture(scope="module")
+    def db_name(self) -> str:
+        return get_env_var("DATABASE_ID", "database name for AlloyDB")
+
+    @pytest.fixture(scope="module")
+    def user(self) -> str:
+        return get_env_var("DB_USER", "user for AlloyDB")
+
+    @pytest.fixture(scope="module")
+    def password(self) -> str:
+        return get_env_var("DB_PASSWORD", "password for AlloyDB")
+
+    @pytest.fixture(scope="module")
+    def iam_account(self) -> str:
+        return get_env_var("IAM_ACCOUNT", "Cloud SQL IAM account email")
+
+    @pytest_asyncio.fixture(scope="module", params=["PUBLIC"])
+    async def engine(
+        self,
+        request,
+        db_project,
+        db_region,
+        db_cluster,
+        db_instance,
+        db_name,
+        user,
+        password,
+    ):
+        engine = AlloyDBEngine.from_instance(
+            project_id=db_project,
+            instance=db_instance,
+            region=db_region,
+            database=db_name,
+            cluster=db_cluster,
+            ip_type=request.param,
+            user=user,
+            password=password,
+        )
+        await aexecute(
+            engine,
+            query=f"CREATE table {COLLECTIONS_TABLE} (uuid VARCHAR, name VARCHAR, cmetadata JSONB)",
+        )
+        await aexecute(
+            engine,
+            query=f"CREATE table {EMBEDDINGS_TABLE} (id VARCHAR, collection_id VARCHAR, embedding vector(768), document TEXT, cmetadata JSONB)",
+        )
+        yield engine
+        await aexecute(engine, f"DROP TABLE {COLLECTIONS_TABLE}")
+        await aexecute(engine, f"DROP TABLE {EMBEDDINGS_TABLE}")
+
+    @pytest.fixture(scope="module")
+    def sample_embeddings(self) -> List[float]:
+        return [0.1] * (VECTOR_SIZE - 1) + [0.2]
```

Comment:
nit: you could use `DeterministicFakeEmbedding(size=VECTOR_SIZE).embed_query("test")` to generate an embedding or you could use `DeterministicFakeEmbedding(size=VECTOR_SIZE)` and the vector store to add embeddings to the table